cfme/tests/services/test_dialog_regex_validation_in_catalog.py

```diff
@@ -66,10 +72,23 @@ def test_dialog_element_regex_validation(appliance, dialog_cat_item):
     catalog_item, ele_name = dialog_cat_item
     service_catalogs = ServiceCatalogs(appliance, catalog_item.catalog, catalog_item.name)
     view = navigate_to(service_catalogs, 'Order')
-    view.fields(ele_name).fill("!@#%&")
-    wait_for(lambda: view.submit_button.disabled, timeout=7)
-    view.fields(ele_name).fill("test_123")
-    wait_for(lambda: not view.submit_button.disabled, timeout=7)
+
+    for value in ["test_123", "!@#$%", ""]:
+        view.fields(ele_name).fill(value)
+        assert view.fields(ele_name).read() == value
+
+        if value in ["!@#$%"]:
+            expected_warning = 'Entered text should match the format: ^([a-z0-9]+_*)*[a-z0-9]+$'
+            assert view.fields(ele_name).input.warning == expected_warning
```

Comment:
No validation that there is no warning for the 'good' value?

Does the warning go away if you replace the bad value with the good? This test will always be putting a good value in with no warning, then putting a bad value in with a warning, but never the opposite order.

I think, since this is a generic "regex_validation" test, it should hit more of these cases, maybe being split into a positive and negative test case function, where negative is testing multiple actions (entering bad value, modifying to another bad value, modifying to a good value, modifying back to a bad value) and validating button state/warning state between each.

At a minimum I would like to see more positive case validation in this test case, the above suggestion about test coverage could be done as a follow-up.